tools/appimage/make-appimage.sh

```diff
@@ -0,0 +1,79 @@
+#! /bin/bash
+
+set -e
+set -o pipefail
+
+show_usage_and_exit() {
+    echo "Usage: $0 <dir>"
+    exit 2
+}
+
+if [[ "$1" == "" ]]; then
+    show_usage_and_exit
+elif [[ ! -e "$1" ]]; then
+    echo "Error: $1 not found"
+    show_usage_and_exit
+elif [[ ! -d "$1" ]]; then
+    echo "Error: $1 is not a directory"
+    show_usage_and_exit
+fi
```

Comment:
Can you add a check that verifies that the script is being executed on the right OS (I guess this is linux-only?) and architecture (x86_64 - we may support more in the future)?

Reply:
I'd rather require an `$ARCH` env var to be set instead, which can be evaluated in the last lines (where appimagetool is called). Detecting the architecture is horribly hard. In a CI script, one can easily add that environment variable manually, though. It's not subject to future changes.

The script can be run on any unix that can execute appimagetool (this includes FreeBSD with the linuxulator, I think). Limiting this to Linux does not make any sense to me. Could you please state how such a check would help you?